### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by assem-ch** [0b01001001/spectree#31]
Thank you for this library, this will be very useful for people who wants to use pydantic in their APIs. 

I  had some issues when I tried the lib:
-  The JSON body is always requested in swagger/redoc in every endpoint even if it's a GET or need no body
- The headers are working for me when I use postman, but in documentation they are shown as an object and they are not passed to endpoints. 
- Can you add option to select what documentation to expose at registration step, and if possible option to change the doc url 
- I guess you are validating the response even if there is no validation is set up : ```The view function did not return a valid response. The return type must be a string, tuple, Response instance, or WSGI callable, but it was a int.```. I think you have to keep everything optional.
- If possible to add description to each endpoint

**Comment by kemingy** [comment]
Thanks for your feedback.

1. I don't really understand this issue. Is it related to #22 ? If not, can you give me some details?
2. I find the Swagger frontend API console has some bugs for `headers`. I'll check this.
3. Currently the config https://github.com/0b01001001/spectree/blob/6abcab8a0946c65975f29702f1fad4ef320b159f/spectree/config.py#L18 support to change the `/apidoc` to other paths. This can be done by `SpecTree('flask', path='doc')`. I can set a default UI in this path that can be changed during registration.
4. The response will be validated only when `resp` is set with `pydantic.BaseModel`. https://github.com/0b01001001/spectree/blob/6abcab8a0946c65975f29702f1fad4ef320b159f/spectree/plugins/flask_plugin.py#L116-L120 https://github.com/0b01001001/spectree/blob/6abcab8a0946c65975f29702f1fad4ef320b159f/spectree/response.py#L25-L29
As for your error message, can you offer the code?
5. The function docs will be used as endpoints' summary(first line) and description(the rest). Try the demo. I guess the tutorial is not good enough.

**Comment by assem-ch** [comment]
1.   yes
<img width="1423" alt="Screen Shot 2020-03-04 at 3 46 18 PM" src="https://user-images.githubusercontent.com/315228/75877015-2c20a700-5e30-11ea-9a67-76ff70797bb3.png">

2. can you present the headers documention as this:
<img width="631" alt="Screen Shot 2020-03-04 at 3 50 53 PM" src="https://user-images.githubusercontent.com/315228/75876983-2034e500-5e30-11ea-8116-555eb14f24c2.png">
4. 
```
[x]   File "/usr/local/lib/python3.7/site-packages/werkzeug/test.py", line 1096, in run_wsgi_app
[x]     app_rv = app(environ, start_response)
[x] TypeError: 'dict' object is not callable
```

**Comment by kemingy** [comment]
1. Is this because of you passed `json` argument in `validate()`? I didn't find this error when I run the demo code in README.

For 1 and 4, can you give me some examples to reproduce this error?

**Comment by assem-ch** [comment]
1. that its for every endpoint , not just json, even GET has it. I am using the version in pypi, may be not the same as master's?
4. try to not set the `resp` argument

**Comment by kemingy** [comment]
For 1, I'll remove JSON body in GET.

For 4, if you are using `flask` as the backend service, you should return what you want. For example: `return jsonify(msg='hello')`. I cannot reproduce this. Can you give me a minimal reproducible example?

**Comment by assem-ch** [comment]
4. Sorry my mistake, I didn't use the jsonify , the lib I was using before was doing json-ification automatically

I can contribute with code , so please just let me if you need any help

**Comment by assem-ch** [comment]
1. please remove json body even in posts when there is no JSON specified. 
6. (new) for the `json` param in validate, what do you think about using the name `body` instead

**Comment by kemingy** [comment]
1. It won't have JSON request body in swagger unless you use `validate(json=MyJSONobject)`. If you still get this error, please offer some example code to reproduce it.
2. JSON is not the only HTTP content-type. So it should be specific as `json`.

**Comment by kemingy** [comment]
The header issue in Swagger page should be fixed in v0.2.8.

**Comment by kemingy** [comment]
I'll close this issue. Feel free to re-open it if you still have the same questions.
